orchagent/orchdaemon.cpp

```diff
@@ -97,7 +97,38 @@ bool OrchDaemon::init()
 
     if (platform == MLNX_PLATFORM_SUBSTRING)
     {
-        m_orchList.push_back(new PfcDurationWatchdog<PfcWdZeroBufferHandler, PfcWdLossyHandler>(m_applDb, pfc_wd_tables));
+
+        static const vector<sai_port_stat_t> portStatIds =
+        {
+            SAI_PORT_STAT_PFC_0_RX_PAUSE_DURATION,
+            SAI_PORT_STAT_PFC_1_RX_PAUSE_DURATION,
+            SAI_PORT_STAT_PFC_2_RX_PAUSE_DURATION,
+            SAI_PORT_STAT_PFC_3_RX_PAUSE_DURATION,
+            SAI_PORT_STAT_PFC_4_RX_PAUSE_DURATION,
+            SAI_PORT_STAT_PFC_5_RX_PAUSE_DURATION,
+            SAI_PORT_STAT_PFC_6_RX_PAUSE_DURATION,
+            SAI_PORT_STAT_PFC_7_RX_PAUSE_DURATION,
+            SAI_PORT_STAT_PFC_0_RX_PKTS,
+            SAI_PORT_STAT_PFC_1_RX_PKTS,
+            SAI_PORT_STAT_PFC_2_RX_PKTS,
+            SAI_PORT_STAT_PFC_3_RX_PKTS,
+            SAI_PORT_STAT_PFC_4_RX_PKTS,
+            SAI_PORT_STAT_PFC_5_RX_PKTS,
+            SAI_PORT_STAT_PFC_6_RX_PKTS,
+            SAI_PORT_STAT_PFC_7_RX_PKTS,
+        };
+
+        static const vector<sai_queue_stat_t> queueStatIds =
+        {
+            SAI_QUEUE_STAT_PACKETS,
```

Comment:
So we don't poll SAI_QUEUE_STAT_PACKETS and SAI_QUEUE_STAT_DROPPED_PACKETS now? Are we going to handle the stats in phase 3?

Reply:
We don't poll them for the purpose of gathering statistics because statistics also depends on platform implementation of handlers. Yes, they will be changed in P3 with a common interface to let different handlers gather statistics in a way that suits them.